File: 小码哥大神5期练习/5、Runtime&RAC&MVVM&新特性/5-06___runtime 关联属性/5-06___runtime 关联属性/ViewController.m
//
//  ViewController.m
//  5-06___runtime 关联属性
//
//  Created by M-coppco on 16/6/18.
//  Copyright © 2016年 M-coppco. All rights reserved.
//

#import "ViewController.h"
#import "NSObject+HJEntension.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSObject *object = [[NSObject alloc] init];
    object.name = @"大大大";
    NSLog(@"%@", object.name);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 小码哥大神5期练习/5、Runtime&RAC&MVVM&新特性/1025/01-Block/01-Block/ViewController.swift
//
//  ViewController.swift
//  01-Block
//
//  Created by M-coppco on 16/7/17.
//  Copyright © 2016年 M-coppco. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    lazy var person: Person = {
        var gtm: Person = Person()
        gtm.eatClosure = {
            print("gtm eat")
        }
        return gtm
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if (self.person.eatClosure != nil) {
            self.person.eatClosure()
        }



        let result = Calculation.makeCalculation { (make: Calculation) -> Void in
            make.a(10).b(20).c(30).d(40)
        }
        print(result)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

